__tests__/app.test.js

```diff
@@ -253,3 +253,21 @@ describe('GET /api/users', () => {
             })
 })
 
+describe('GET /api/articles (topic query)', () => {
+    test('Responds with a status 200 containing an array of objects, each containing the desired topic', () => {
+        return request(app).get('/api/articles?topic=cats').expect(200)
+            .then(({body}) => {
+                expect(Array.isArray(body)).toBe(true);
+                body.forEach((article) => {
+                    expect(article.topic).toBe('cats');
+                })
+            })
+    })
+    test('Responds with a status 404 containing an appropriate message when no articles are found of requested topic', () => {
+        return request(app).get('/api/articles?topic=dogs').expect(404)
+            .then(({body}) => {
+                expect(body.msg).toContain('No articles found for this topic:');
+            })
+    })
+})
+
```

Comment:
Good testing so far, but we need to do a little bit more consideration here, think about the following...
What if a topic doesn't exist? We'd expect a 404 here
What if a topic does exist but doesn't have any articles? This should be a 200

We'd need to query the topics table to find out if what has been passed in is valid or notes then use that info to determine our response, give this a shot!